components/mailgun/mailgun.app.js

```diff
@@ -0,0 +1,155 @@
+const axios = require("axios");
+const get = require("lodash/get");
+const retry = require("async-retry");
+
+module.exports = {
+  type: "app",
+  app: "mailgun",
+  propDefinitions: {
+    domain: {
+      type: "string",
+      label: "domain",
+      description: "Domain used to send and receive email.",
+      async options(context) {
+        const q = context.query;
+        const options = [];
+        const results = await this.getAllDomains();
+        for (const domains of results) {
+          options.push({
+            label: domains.name,
+            value: domains.name,
+          });
+        }
```

Comment:
FYI, if label and value are the same, you can just create an array of the domain names instead of an array of label/value pairs.
```
for (const domains of results) {
  options.push(domains.name);
}
```